### ISSUE TITLE
Same UUID as another bridged accessory

### ISSUE THREAD

**Issue by S1rPsych0** [hjdhjd/homebridge-myq2#3]
Plugin works great until a reboot. When launching Homebridge after reboot, the error following is displayed:

[MyQ2] Error polling MyQ servers: Error: Cannot add a bridged Accessory with the same UUID as another bridged Accessory: 

and Homebridge fails to start

**Comment by DoctorOgg** [comment]
Yup!, same issues for me as well

**Comment by stalinCCCP415** [comment]
Same issue here.

**Comment by hjdhjd** [comment]
Shutdown homebridge. Try deleting the persist folder and then restart. If that doesn’t work, delete the persist and accessories folders and then start homebridge. Please let me know if that does the trick.

**Comment by stalinCCCP415** [comment]
What works for me every time is deleting the cahcedaccessories file from Accessories, then restarting the container. On the next homebridge startup, same problem, same routine. Looks like the same issue is happening with other myq plugins (liftmaster2).

**Comment by S1rPsych0** [comment]
@stalinCCCP415 , yeah that's the only thing I've found that works. I created a script that deletes the cahcedaccessories file, then launches HomeBridge on startup. The problem with that is Home Kit sees my garage door as a new accessory  after any restart. Any settings or automations have to be reconfigured.

**Comment by dppeak** [comment]
Was this fixed in the latest release of Liftmaster2?

**Comment by DoctorOgg** [comment]
leaving the thread as i have given up and replaced with another solution

**Comment by Dirtrunner702** [comment]
Should we move onto another plugin? The issue continues to occur for me too. I also have 2 separate garage controllers configured. 
![image](https://user-images.githubusercontent.com/25218695/66353960-3a69a600-e918-11e9-847d-10b7d9cfd3cb.png)

**Comment by bunchjesse** [comment]
The problem is how we're saving and loading the accessory from the cache. Homebridge expects the `accessories` property to be an array of accessories. We're currently storing these as a dictionary and they're not round-tripping correctly.

So this is failing:

`var accessory = self.accessories[thisDeviceID];`

**Comment by bunchjesse** [comment]
Scratch that...looks like Homebridge isn't properly saving the serialized `accessory.context`?

**Comment by bunchjesse** [comment]
Alright. Figured it out. The `accessory.context` is not being written out to the accessory cache. This happens because the accessory cache is only written when registering or updating accessories.

We either need to a) delay calling `registerPlatformAccessories` until after we build up the `accessory.context` or call `updatePlatformAccessories` once we're done.

I added the following after line 386 to trigger a flush of the cache:

```
// Ensure the cache is up to date
self.api.updatePlatformAccessories([accessory]);
```
